google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -1035,7 +1035,8 @@ public static BlobListOption fields(BlobField... fields) {
       EXT_HEADERS,
       SERVICE_ACCOUNT_CRED,
       SIGNATURE_VERSION,
-      HOST_NAME
+      HOST_NAME,
+      VIRTUAL_HOST_NAME
```

Comment:
@houglum, instead of creating a new option. Could you update the generated URL instead to use the virtual host name?

If it's considered better practice to use `bucket.storage.googleapis.com/`, I'd push for implementations to use this format instead.

Reply:
Do you think changing the resulting URLs might break some users, e.g. those who have special networking rules set up?  What if a user is utilizing VPC and has a rule to allow requests with the host set to `storage.googleapis.com`?  Suddenly, their requests to signed URLs begin to fail because they're using the host `somebucket.storage.googleapis.com`.

I do agree that it's better practice to use the virtual-hosted style of URL though.  If you think the potential to break existing users is minimal, I'd be fine with changing the default URL format to virtual-hosted style.  If we do that, do we want to add an option to allow people to use the old path-style format?

Reply:
FYI for those following along: We've written up a doc to summarize our options and get input on what's the most desirable approach here.